pkg/schedule/schedulers/balance_leader.go

```diff
@@ -79,7 +79,7 @@ func (conf *balanceLeaderSchedulerConfig) update(data []byte) (int, any) {
 			}
 			return http.StatusBadRequest, "invalid batch size which should be an integer between 1 and 10"
 		}
-		if err := conf.persistLocked(); err != nil {
+		if err := conf.save(conf); err != nil {
```

Comment:
how about conf.save()

Reply:
If we change it to the form of `conf.save()`, then we will have to save `conf` as a pointer(any) in baseSchedulerConfig. I'm not sure if this is a good idea. If everyone agrees, I can make this change.

cc @rleungx

Reply:
I've made this change, we can compare them and choose the best one. https://github.com/tikv/pd/pull/8515/commits/96ef87e77ce33ca3e2a860c1715780845a038a69

Actually, I also think that `save()` is better👀